TeamCode/src/main/java/org/firstinspires/ftc/teamcode/AprilTagDetectionFrameProcessor.java

```diff
@@ -83,31 +88,25 @@ public AprilTagDetectionPipeline(double tagsize, double fx, double fy, double cx
     }
 
     @Override
-    public void finalize()
-    {
+    protected void finalize() {
         // Might be null if createApriltagDetector() threw an exception
-        if(nativeApriltagPtr != 0)
-        {
+        if (nativeApriltagPtr != 0) {
             // Delete the native context we created in the constructor
             AprilTagDetectorJNI.releaseApriltagDetector(nativeApriltagPtr);
             nativeApriltagPtr = 0;
-        }
-        else
-        {
+        } else {
             System.out.println("AprilTagDetectionPipeline.finalize(): nativeApriltagPtr was NULL");
         }
     }
 
     @Override
-    public Mat processFrame(Mat input)
-    {
+    public void processFrame(Mat input, Mat output) {
```

Comment:
Our FrameProcessor passes an `output` mat to write to instead of just returning the same `input` one.  Otherwise, pretty much everything in this function is the same.